trigger.py

```diff
@@ -6,91 +6,87 @@
+        self.trig_ID = ""
+        self.trig_type = self.TYPES[self.TYPES["Default"]]
+        self.edge = self.EDGES[self.EDGES["Default"]]
+        self.level = self.LEVELS[self.LEVELS["Default"]]
+
+        super().__init__(node)
+
+    def load_xml(self, node: ET.Element):
         """
         re-initialize attributes for existing Trigger from children of node. 
         'node' is of type xml.etree.ElementTree.Element, with tag="trigger"
         """
+
         for child in node:
-            
+
             if type(child) == ET.Element:
-            
-                if child.tag == "id": 
-                    self.trigID = child.text # script trigger 0
+
+                if child.tag == "id":
+                    self.trig_ID = child.text  # script trigger 0
 
                 elif child.tag == "source":
-                    self.source = child.text # PFI 0
+                    self.source = child.text  # PFI 0
 
                 elif child.tag == "type":
-
-                    if child.text == "level":
-                        self.trigType = self.trigTypes["Level"]
-
-                    # else, stick with default initialization
+                    self.set_by_dict("trig_type", child.text, self.TYPES)
 
                 elif child.tag == "edge":
-
-                    if child.text == "Falling Edge":
-                        self.edge = self.edges["Falling Edge"]
-                        pass
-
-                    # else, stick with default initialization
+                    self.set_by_dict("edge", child.text, self.EDGES)
 
                 elif child.tag == "level":
+                    self.set_by_dict("level", child.text, self.LEVELS)
 
-                    if child.text == "Low Level":
-                        self.level = self.levels["Low Level"]
-                        pass
-                    
                 else:
-                    # TODO: replace with logger
-                    print("Not a valid trigger attribute") 
-        
-    def __repr__(self): # mostly for debugging
-        return (f"Trigger(id={self.trigID}, source={self.source}, "
-                f"type={self.trigType}, edge={self.edge}, level={self.level})")
-                
+                    self.logger.warning(f"{child.tag} is not a valid trigger attribute")
 
-# with some refactoring, this could inherit from Trigger
-class StartTrigger: 
+    def __repr__(self):  # mostly for debugging
+        return (f"Trigger(id={self.trig_ID}, source={self.source}, "
+                f"type={self.trig_type}, edge={self.edge}, level={self.level})")
+                
 
-    # TODO add Default key
-    edges = {"Rising Edge": 12,
-             "Falling Edge": 13}
+class StartTrigger(XMLLoader):
+    """
+    TODO : @Preston write docstring for this class
+    """
+    EDGES = {"Rising Edge": 12,
+             "Falling Edge": 13,
+             "Default": "Rising Edge"}
 
-    nidaqmx_edges = {"Rising" : Edge.RISING,
+    # TODO : This is unused... Remove? -Juan
```

Comment:
This is used in AnalogInput, among other places where NIDAQmx triggering occurs. So please don't remove :)